### PR TITLE
Update toolchain and packages to build with gcc 11.2.0 and glibc 2.34

### PR DESCRIPTION
**Pull request by anphel31**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?

Update toolchain to build with the latest compiler (gcc 11.2.0) and C library (glibc 2.34). Update other toolchain packages as required to build the new compiler and glibc. Also modify and update packages as needed to properly build with the updated toolchain.
See: https://gcc.gnu.org/gcc-11/porting_to.html and https://gcc.gnu.org/gcc-10/porting_to.html for information about porting to the new gcc11.

###### Change Log  

- Update gcc to version 11.2.0
- Update glibc to version 2.34
- Update toolchain to bootstrap and compile new gcc/glibc. Add patches to toolchain scripts as required. Update temp toolchain package versions to SPEC versions (file, bzip2, perl, rpm). Remove unnecessary compilation of zstd in temp toolchain.
- Set -fcommon in global compiler flags, since gcc11 defaults to -fno-common which causes many breaks with legacy C code. See here for more details: https://gcc.gnu.org/gcc-10/porting_to.html
- Patch kernel to build with gcc11, and bump kernel package release numbers. Update kernel config files for gcc 11.2.0 and binutils 2.37 versions. Also update configs per results of local builds.
- Modifed (but did not upgrade) the following packages to build with gcc11/glibc 2.34: auoms, audit, azure-iot-sdk-c, blobfuse, cpio, device-mapper-multipath, dhcp, efivar, fuse, gpgme, libacvp, m4, make, nss, openjdk8_aarch64, qemu-kvm, qt5-qtbase, syslinux
- Remove packages: ctags, tboot, tcp_wrappers
- Temporarily disable packages (which use openjdk8 and get stuck compiling in docker): ant, ant-contrib, jna, R, conda, pycostat, python-pycosat
- Set "GODEBUG=netdns=go" when running srpmpacker to use go for DNS resolution instead of libc. This resolves error "dial tcp: lookup cblmarinerstorage.blob.core.windows.net: device or resource busy" seen when packing SRPMS in chroot. See https://go-review.googlesource.com/c/go/+/11584/
- Update autofs to version 5.1.8
- Update azure-iotedge to version 1.1.8
- Update binutils to version 2.37
- Update bison to version 3.7.6
- Update catch to version 2.13.7
- Update coreutils to version 8.32
- Update diffutils to version 3.8
- Update dtc to version 1.6.1
- Update eflutils to version 0.185
- Update findutils to version 4.8.0
- Update gawk to version 5.1.0
- Update gdb to version 11.1
- Update gdbm to version 1.21
- Update gmp to version 6.2.1
- Update grep to version 3.7
- Update grpc to version 1.41.1
- Update gzip to version 1.11
- Update ipxe to version 1.21.1
- Update kbd to version 2.2.0
- Update libcomps to version 0.1.18
- Update libgpg-error to version 1.43
- Update libiothsm-std to version 1.1.8
- Update libmpc to version 1.2.1
- Update libtirpc to version 1.3.2
- Update lldpad to version 1.1.0
- Update lttng-consume to version 0.2.1
- Update mpfr to version 4.1.0
- Update nfs-utils to version 2.5.4
- Update nlohmann-json to version 3.10.4
- Update ntp to version 4.2.8p15
- Update openjdk8 (x86_64 only) to version 1.8.0.302
- Update rocksdb to version 6.26.0
- Update sed to version 4.8
- Update tar to version 1.34
- Update texinto to version 6.8
- Update toml11 to version 3.7.0
- Update util-linux to version 2.37.2
- Update valgrind to version 3.18.1
- Update xz to version 5.2.5

###### Does this affect the toolchain?  

**YES**

###### Links to CVEs  
- https://nvd.nist.gov/vuln/detail/CVE-YYYY-XXXX

###### Test Methodology

- Pipeline build id: 
- (AMD64 toolchain) 147133
- (AMD64 RPMS) 147151
- (ARM64 toolchain) 146349
- (ARM64 RPMS) 147293

### CODE REVIEW

File: SPECS/sed/sed.spec
Comment (PawelWMS) [inline SPECS/sed/sed.spec:19]: Do you know, if we could use `>= %{version}` instead?

Comment (anphel31) [reply SPECS/sed/sed.spec:19]: Yes I believe so, will update

File: SPECS/perl/perl.spec
Comment (PawelWMS) [inline SPECS/perl/perl.spec:3]: Not sure if we should mention infrastructure-related reasons in the spec comments. Also, shouldn't we mention this change in the changelog?

Comment (anphel31) [reply SPECS/perl/perl.spec:3]: Yes the changelog should be updated if this change stays. But for now it's a temporary change intended to unblock the build pipeline, which has been getting stuck on this package.

**Review by nicogbg — APPROVED** [review]

File: SPECS/kernel-hyperv/config
Comment (christopherco) [inline SPECS/kernel-hyperv/config:720]: The GCC plugins missing/disabled is odd to see. We use these plugins for hardening the kernel. Not necessarily a blocker for the merge but something we need to revisit

File: SPECS/kernel-hyperv/kernel-hyperv.spec
Comment (christopherco) [inline SPECS/kernel-hyperv/kernel-hyperv.spec:96]: This should be a patch rather than an inline sed command in the spec.

File: SPECS/kernel/kernel.spec
Comment (christopherco) [inline SPECS/kernel/kernel.spec:329]: commenting again that this should be a patch file instead of a sed.

File: SPECS/libacvp/libacvp.spec
Comment (christopherco) [inline SPECS/libacvp/libacvp.spec:38]: Revert this line. The original CFLAGS is important to preserve because it enables specific OpenSSL vectors needed for libacvp testing.

**Review by christopherco — CHANGES_REQUESTED** [review]
Reviewed the kernel-headers, kernel-hyperv, kernel, and libacvp specs. The specific change requested is libacvp. It'd be good to get the kernel comments addressed too but may be okay to address in a future PR.

File: SPECS/libacvp/libacvp.spec
Comment (PawelWMS) [reply SPECS/libacvp/libacvp.spec:38]: But I wouldn't want to lose our global `CFLAGS` values. Would something like this be acceptable instead:

```suggestion
    CFLAGS="$CFLAGS -pthread -DACVP_NO_RUNTIME -DOPENSSL_KWP -DOPENSSL_KDF_SUPPORT -O0 -g" \
```

File: SPECS/kernel-hyperv/kernel-hyperv.spec
Comment (anphel31) [reply SPECS/kernel-hyperv/kernel-hyperv.spec:96]: Sure, will convert to patch

File: SPECS/kernel/kernel.spec
Comment (anphel31) [reply SPECS/kernel/kernel.spec:329]: Sure, will convert to patch

File: SPECS/kernel-hyperv/config
Comment (anphel31) [reply SPECS/kernel-hyperv/config:720]: I'm not sure yet why these are turned off with the new compiler. Definitely will follow up though to re-enabled after merging

File: SPECS/libacvp/libacvp.spec
Comment (christopherco) [reply SPECS/libacvp/libacvp.spec:38]: We would need to confirm whether the resulting binary still does what we expect it to do with the $CFLAGS added in. It would likely be fine but without confirming, I am of the opinion that it is best to keep the original CFLAGS override until we can confirm adding $CFLAGS is truly safe.

Comment (anphel31) [reply SPECS/libacvp/libacvp.spec]: I've reverted the change here

File: SPECS/kernel-hyperv/kernel-hyperv.spec
Comment (anphel31) [reply SPECS/kernel-hyperv/kernel-hyperv.spec:96]: Will fix post merge

File: SPECS/kernel/kernel.spec
Comment (anphel31) [reply SPECS/kernel/kernel.spec:329]: Will fix post merge

**Review by christopherco — APPROVED** [review]

File: SPECS/kernel-hyperv/kernel-hyperv.spec
Comment (anphel31) [reply SPECS/kernel-hyperv/kernel-hyperv.spec]: Converted to patch

File: SPECS/kernel/kernel.spec
Comment (anphel31) [reply SPECS/kernel/kernel.spec]: Converted to patch

**Pull request merged**
